plugins/builds/create.js

```diff
@@ -16,18 +18,49 @@ module.exports = (server) => ({
             scope: ['user']
         },
         handler: (request, reply) => {
-            const Build = new Model.Build(
+            const build = new Model.Build(
                 server.settings.app.datastore,
-                server.settings.app.executor
+                server.settings.app.executor,
+                options.password
             );
             const payload = {
                 jobId: request.payload.jobId,
                 apiUri: request.server.info.uri,
                 tokenGen: (buildId) =>
                     request.server.plugins.login.generateToken(buildId, ['build'])
             };
+            const username = request.auth.credentials.username;
 
-            Build.create(payload, (err, data) => {
+            // Check if user can create a build
+            async.waterfall([
+                // Get job
+                (next) => build.job.get(request.payload.jobId, next),
+
+                // Get pipeline
+                (jobObj, next) => build.pipeline.get(jobObj.pipelineId, next),
+
+                // Get permissions
+                (pipelineObj, next) => build.user.getPermissions({
+                    username,
+                    scmUrl: pipelineObj.scmUrl
+                }, next),
+
+                // eslint-disable-next-line consistent-return
+                (permissions, next) => {
+                    if (!permissions.push) {
+                        return reply(boom.unauthorized(`User ${username} `
+                          + 'does not have push permission for this repo'));
+                    }
+                    next();
```

Comment:
Remove the lint error by adding `return` in front of this next (or wrap in else)